### ISSUE TITLE
Playback not resuming

### ISSUE THREAD

**Issue by lvitalis** [MoleMan1024/audiowagon_beta#16]
**Describe the issue**
We re-entering the car after some time, playback does not resume as it should.

**To Reproduce**
- Enter the car
- Accept the USB permission
- Start a track
- Exit and lock the car
- Wait at least 20 seconds
- Unlock and re-enter the car
- Accept the USB permission
- Music does not resume

`Please note that if you unlock and re-enter the car quickly (but slow enough that music stops and screen turns off), the USB permission is not requested, the playback resumes.`

**Expected behavior**
Last track should resume where it was paused.

**AudioWagon version**
0.25

**Logfiles**
Log of two successful resumes
[Two successful resumes.log](https://github.com/MoleMan1024/audiowagon_beta/files/7178866/Two.successful.resumes.log)

Log of a failed resume
[One failed resume.log](https://github.com/MoleMan1024/audiowagon_beta/files/7178865/One.failed.resume.log)

**Comment by lvitalis** [comment]
I suggest that the car firmware version be a part of the issue template, under AudioWagon version.
I forgot to mention it.

**Comment by MoleMan1024** [comment]
Ok as I suspected we receive `onPrepare()` and `onPlay()`, this comes 3 seconds after intent to turn on the screen.

Two.successful.resumes.log
```
2021-09-16 10:41:26.017 [DEBUG   ][6715 ][PowerEvtReceiver    ] Received notification: Intent { act=android.intent.action.SCREEN_ON flg=0x50200010 }
2021-09-16 10:41:29.176 [DEBUG   ][6715 ][AudioSessCB         ] onPlay()
...
2021-09-16 10:41:59.657 [DEBUG   ][6715 ][PowerEvtReceiver    ] Received notification: Intent { act=android.intent.action.SCREEN_ON flg=0x50200010 }
2021-09-16 10:42:02.480 [DEBUG   ][6715 ][AudioSessCB         ] onPlay()
```

The bug report said this works only when doing it quickly. Likely the Android system is not put to sleep yet when you exit and re-enter the car quickly, it will just turn off the screen.

The other log ...

I only see that (probably) the user ejected the USB drive here
```
2021-09-16 10:43:05.153 [DEBUG   ][6715 ][AudioSessCB         ] onCustomAction(de.moleman1024.audiowagon.ACTION_EJECT, Bundle[{}])
```
(some duplicates of that message later, need to check that)

But no media session callback was called before that. Either it was not sent, or we missed it because of late logging/app startup or... something else.

@lvitalis : does resuming the audio work fine for other music apps when you leave the car for a longer time?

**Comment by lvitalis** [comment]
> does resuming the audio work fine for other music apps when you leave the car for a longer time?

Yes, I can confirm that Spotify, YouTube Music resume all the time. Bluetooth and FM radio also.

**Comment by MoleMan1024** [comment]
I have implemented the latch function for onPlay(), hopefully this should now also work for AudioWagon. Will be in next version.
